referenceframe/worldstate.go

```diff
@@ -64,45 +75,44 @@ func WorldStateToProtobuf(worldState *WorldState) (*commonpb.WorldState, error)
 	}
 
 	return &commonpb.WorldState{
-		Obstacles:  convertGeometriesToProto(worldState.Obstacles),
+		Obstacles:  convertGeometriesToProto(worldState.obstacles),
 		Transforms: transforms,
 	}, nil
 }
 
 // ObstaclesInWorldFrame takes a frame system and a set of inputs for that frame system and converts all the obstacles
 // in the WorldState such that they are in the frame system's World reference frame.
 func (ws *WorldState) ObstaclesInWorldFrame(fs FrameSystem, inputs map[string][]Input) (*GeometriesInFrame, error) {
-	transformGeometriesToWorldFrame := func(gfs []*GeometriesInFrame) (*GeometriesInFrame, error) {
-		nameCheck := make(map[string]bool)
-		allGeometries := make([]spatial.Geometry, 0, len(gfs))
-
-		unnamedCount := 1
-
-		for _, gf := range gfs {
-			tf, err := fs.Transform(inputs, gf, World)
-			if err != nil {
-				return nil, err
-			}
-			for _, g := range tf.(*GeometriesInFrame).Geometries() {
-				geomName := g.Label()
-				if geomName == "" {
-					geomName = unnamedWorldStateGeometryPrefix + strconv.Itoa(unnamedCount)
-					g.SetLabel(geomName)
-					unnamedCount++
-				}
-
-				if _, present := nameCheck[geomName]; present {
-					return nil, fmt.Errorf("cannot specify multiple geometries with the same name: %s", geomName)
```

Comment:
there no longer is a need to check this when we transform the geometries if we always create worldstates with good naming systems!